app/Http/Controllers/Legacy/Web/ActionsController.php

```diff
@@ -77,6 +77,28 @@ public function store(Request $request)
         return redirect()->route('campaign-ids.show', $request['campaign_id']);
     }
 
+    /**
+     * Edit an existing action.
+     */
+    public function edit($campaignId, $actionId)
+    {
+        $action = Action::find($actionId);
+
+        $postTypes = [
+            'text',
+            'photo',
+            'voter-reg',
+            'share-social',
+            'phone-call',
+        ];
```

Comment:
I wonder if there's a way to save this list somewhere so it doesn't have to be listed out manually in more than once place?

Reply:
hm interesting! I agree this is a little weird to hardcode but not sure where would be a better way to access this list?

Reply:
@chloealee yup I wasn't sure either, I was just thinking outloud!

Reply:
Could it make sense to add a `PostType` class similar to the [`ImportType` in Chompy](https://github.com/DoSomething/chompy/blob/master/app/Import.php#L33), or [`PasswordResetType` in Northstar](https://github.com/DoSomething/northstar/blob/509dcee48431b8283d9b436283d5a14a45139fec/app/PasswordResetType.php#L26)?

Reply:
ooo that is really cool @aaronschachter ! So would this work by creating a `PostType` class and then pass `$type` to the view similar to how you did this [here](https://github.com/DoSomething/chompy/pull/59/files#diff-e489ef4664b1c24151b9b01896103403R77)? Where does this `$type` variable come from?

Reply:
That `$type` variable is [set in the same file](https://github.com/DoSomething/chompy/blob/ae50b2f567e43458f0a35d46a4579260a22656da/app/Http/Controllers/ImportController.php#L60), saving the `type` query string parameter... but that's just specific to the `ImportController`. The intent for the `ImportType` and `PasswordResetType` classes I linked to is to DRY hardcoding strings like `rock-the-vote` all over the codebase, but instead calling `ImportType::rockTheVote()`. The `all()` function returns all valid import types. 

So for a PostType, we could define different readable functions like `voterRegistration()`, `text()`, `photo()`, `phoneCall()`, `socialShare()`, that return the string values we save in the database.. and an `all` function should return all of these string values, so this edit function could be refactored as:
```
    public function edit($campaignId, $actionId)
    {
        return view('actions.edit')->with([
            'campaignId' => $campaignId,
            'action' => Action::find($actionId),
            'postTypes' => PostTypes::all(),
        ]);
 }
```

Reply:
Not within scope of this PR, but we could eventually start replacing hardcoded post type strings with the newer `PostType` functions... which could help prevent typos for string values, or make it easy to change these values if we ever needed to

e.g.  [`getStatusRules`](https://github.com/DoSomething/rogue/blob/master/app/Http/Requests/PostRequest.php#L69) would look like:
```
    private function getStatusRules($type)
    {
        switch ($type) {
            PostType::photo():
                $rule = 'in:pending,accepted,rejected';
                break;
            case PostType::voterRegistration():
                $rule = 'in:pending,register-form,register-OVR,confirmed,ineligible,uncertain';
                break;
            case PostType::phoneCall():
                $rule = 'in:accepted,incomplete';
                break;
            default:
                $rule = 'in:pending,accepted,rejected';
        }
        return $rule;
```

Related - we could also add a helper for PostStatus

Reply:
Ah super smart - thanks for explaining more, @aaronschachter ! Updated this :)